code/modules/library/lib_items.dm

```diff
@@ -18,106 +18,123 @@
+	else if(istype(O, /obj/item/weapon/crowbar) && user.a_intent == I_HELP) //Only way to deconstruct, needs help intent
+		playsound(get_turf(src), 'sound/items/Crowbar.ogg', 75, 1)
+		user.visible_message("<span class='warning'>[user] starts disassembling \the [src].</span>", \
+		"<span class='notice'>You start disassembling \the [src].</span>")
+		busy = 1
+
+		if(do_after(user, src, 50))
+			playsound(get_turf(src), 'sound/items/Deconstruct.ogg', 75, 1)
+			user.visible_message("<span class='warning'>[user] disassembles \the [src].</span>", \
+			"<span class='notice'>You disassemble \the [src].</span>")
+			busy = 0
 			getFromPool(/obj/item/stack/sheet/wood, get_turf(src), 5)
-			density = 0
-			qdel(src)
+			Destroy(src)
+			return
+		else
+			busy = 0
 		return
 	else if(istype(O, /obj/item/weapon/wrench))
-		playsound(src.loc, 'sound/items/Ratchet.ogg', 50, 1)
-		user << (anchored ? "<span class='notice'>You unfasten the [src] from the floor.</span>" : "<span class='notice'>You secure the [src] to the floor.</span>")
 		anchored = !anchored
+		playsound(get_turf(src), 'sound/items/Ratchet.ogg', 50, 1)
+		user.visible_message("<span class='warning'>[user] [anchored ? "":"un"]anchors \the [src] [anchored ? "to":"from"] the floor.</span>", \
+		"<span class='notice'>You [anchored ? "":"un"]anchor the [src] [anchored ? "to":"from"] the floor.</span>")
 	else if(istype(O, /obj/item/weapon/pen))
-		var/newname = stripped_input(usr, "What would you like to title this bookshelf?")
+		var/newname = stripped_input(user, "What category title would you like to give to this [name]?")
 		if(!newname)
 			return
 		else
 			name = ("bookcase ([sanitize(newname)])")
+	else if(O.damtype == BRUTE || O.damtype == BURN)
+		user.delayNextAttack(10) //We are attacking the bookshelf
+		health -= O.force
+		user.visible_message("<span class='warning'>\The [user] hits \the [src] with \the [O].</span>", \
+		"<span class='warning'>You hit \the [src] with \the [O].</span>")
+		healthcheck()
+		return
+
 	else
-		switch(O.damtype)
-			if("fire")
-				src.health -= O.force * 1
-			if("brute")
-				src.health -= O.force * 0.75
-			else
-		if (src.health <= 0)
-			visible_message("<span class=warning>The bookcase is smashed apart!</span>")
-			getFromPool(/obj/item/stack/sheet/wood, get_turf(src), 3)
-			qdel(src)
-		..()
+		..() //Weapon checks for weapons without brute or burn damage type and grab check
 
 /obj/structure/bookcase/attack_hand(var/mob/user as mob)
 	if(contents.len)
-		var/obj/item/weapon/book/choice = input("Which book would you like to remove from the shelf?") as null|obj in contents
+		var/obj/item/weapon/book/choice = input("Which book would you like to remove from \the [src]?") as null|obj in contents
 		if(choice)
-			if(!usr.canmove || usr.stat || usr.restrained() || !in_range(loc, usr))
+			if(user.restrained() || user.stat || user.weakened || user.stunned || user.paralysis || user.resting || get_dist(user, src) > 1)

```

Comment:
Just make the `attack_hand()` sane properly?

It's exactly the same as making `topic()` sane.